gdal/apps/gdal_translate_lib.cpp

```diff
@@ -1970,10 +1970,15 @@ static void CopyBandInfo( GDALRasterBand * poSrcBand, GDALRasterBand * poDstBand
         if (poSrcBand->GetDefaultRAT() && bCopyRAT)
         {
             GDALRasterAttributeTable *poNewRAT = poSrcBand->GetDefaultRAT()->Clone();
+
+            // strip histogram data (as definied by the source RAT)
             poNewRAT->RemoveStatistics();
             if( poNewRAT->GetColumnCount() )
             {
                 poDstBand->SetDefaultRAT( poNewRAT );
+
+                // strip histogram data (as definied by the destination RAT)
+                poDstBand->GetDefaultRAT()->RemoveStatistics();
```

Comment:
are you sure this is needed ? poDstBand here is a VRTRasterBand. If the statistics have been removed from poNewRat, there is no reason for them to be present in poDstBand->GetDefaultRAT() at that point

Reply:
poDstBand is typically a DefaultRasterAttributeTable... but it only gets converted to (eg.) a HFARasterAttributeTable _after_ it's assigned to the new band. So the DefaultRasterAttributeTable::RemoveStatistics() would need to remove the histogram even though it probably shouldn't know about it yet? And the HFARasterAttributeTable::RemoveStatistics() is never called to clean itself up.

As it happens all the existing remove-things logic sits in the DefaultRasterAttributeTable _anyway_, so it's somewhat academic.

Happy to strip it out.